File: docs/research/video-tests/01-release/scripts/prepare.mjs
import fs from 'node:fs';
import {execFileSync} from 'node:child_process';
const r=JSON.parse(fs.readFileSync('evidence/run.json','utf8'));
const v=JSON.parse(fs.readFileSync('evidence/github-verification.json','utf8'));
if(r.run.state!=='completed'||!r.run.finalText.includes(v.tag_name))throw Error('Run did not independently match GitHub release');
if(r.browser.url!==v.html_url)throw Error('Release page was not left visible');
const startMs=r.markers.find(m=>m.name==='start_click').atMs;
const done=r.events.find(e=>e.event.type==='run.state'&&e.event.state==='completed');
const taskSeconds=(done.atMs-startMs)/1000;
let active=false;const stages=[{atMs:startMs,label:'Gemini plans the next step.'}];
for(const {atMs,event:e}of r.events){
 if(e.type!=='run.tool')continue;
 if(e.name==='browser_task'&&e.status==='executing'){active=true;stages.push({atMs,label:'Gemini delegates the browser subtask to Jev.'});}
 if(e.name==='browser_task'&&e.status==='done'){active=false;stages.push({atMs,label:r.run.jev?.fallbacks?'Jev asks for help. Gemini takes over.':'Gemini reads the evidence and plans what comes next.'});}
 if(!active&&e.name==='browser_batch'&&e.status==='executing')stages.push({atMs,label:'Gemini uses a planned browser action.'});
 if(!active&&e.name==='browser_read'&&e.status==='executing')stages.push({atMs,label:'Gemini reads the page to check the answer.'});
}
const data={version:v.tag_name,releaseDate:new Date(v.published_at).toLocaleDateString('en-US',{month:'long',day:'numeric',year:'numeric',timeZone:'UTC'}),source:v.html_url,startMs,taskSeconds,stages:stages.filter((s,i)=>!stages[i+1]||stages[i+1].atMs-s.atMs>=700),prompt:r.prompt,recordedAt:r.captureDate,model:r.run.model??'google/gemini-3.8-flash',jev:r.run.jev,costUsd:r.run.costUsd};
fs.writeFileSync('src/run-data.json',JSON.stringify(data,null,2));
execFileSync('ffmpeg',['-hide_banner','-loglevel','error','-y','-ss',String((r.markers.find(m=>m.name==='capture_end').atMs-1800)/1000),'-i','public/capture.mp4','-vf','crop=876:548:124:300','-frames:v','1','public/release.png']);
console.log(JSON.stringify(data,null,2));

File: docs/research/video-tests/02-research-file/scripts/prepare.py
from pathlib import Path
import json,re,hashlib,sqlite3,subprocess
r=json.loads(Path('evidence/run.json').read_text());assert r['run']['state']=='completed'
report=Path.home()/'linux-agent-tutorial-research-workspace/python-cli-tools.md'
s=report.read_text();urls=['https://docs.astral.sh/uv/guides/tools/','https://pipx.pypa.io/stable/']
assert all(u in s for u in urls);assert 'uvx ruff' in s and 'pipx run ruff' in s
assert len(s.split())<180
start=next(m['atMs'] for m in r['markers'] if m['name']=='start_click')
done=next(e['atMs'] for e in r['events'] if e['event'].get('type')=='run.state' and e['event'].get('state')=='completed')
stages=[{'atMs':start,'label':'Auto chooses a planned approach.'}];nav=[]
for e in r['events']:
 v=e['event']
 if v.get('type')!='run.tool' or v.get('status')!='executing':continue
 p=v.get('preview','')
 if v['name']=='browser_act' and p.startswith('navigate '):
  nav.append(p[9:]);stages.append({'atMs':e['atMs'],'label':'Gemini reads the uv documentation.' if 'astral.sh' in p else 'Gemini reads the pipx documentation.'})
 if v['name']=='terminal_input':stages.append({'atMs':e['atMs'],'label':'Gemini writes the note, then displays the saved file.'})
assert all(u in nav for u in urls)
rows=[[c.strip() for c in l.strip().strip('|').split('|')] for l in s.splitlines() if l.startswith('|')]
data={'startMs':start,'taskSeconds':(done-start)/1000,'stages':stages,'prompt':r['prompt'],'recordedAt':r['captureDate'],'costUsd':r['run']['costUsd'],'jev':r['run']['jev'],'title':s.splitlines()[0].lstrip('# '),'intro':s.split('\n\n')[1],'headers':rows[0],'rows':rows[2:],'sources':urls,'reportSha256':hashlib.sha256(s.encode()).hexdigest(),'wordCount':len(s.split()),'viewSwitchMs':next(m['atMs'] for m in r['markers'] if m['name']=='view_terminal')}
Path('src/run-data.json').write_text(json.dumps(data,indent=2));Path('public/python-cli-tools.md').write_text(s)
Path('output/python-cli-tools.md').write_text(s)
c=sqlite3.connect(Path.home()/'.local/share/linux-agent-workbench-tutorial-research/state.sqlite')
events=c.execute("select ts,type,payload_json from run_events where run_id=? and type in ('browser.route','browser.task')",(r['run']['runId'],)).fetchall()
Path('evidence/route-events.json').write_text(json.dumps([{'ts':t,'type':typ,'payload':json.loads(p)} for t,typ,p in events],indent=2))
for name,t in [('uv',12.1),('pipx',15.3)]:
 subprocess.run(['ffmpeg','-hide_banner','-loglevel','error','-y','-ss',str(t),'-i','public/capture.mp4','-vf','crop=876:548:124:300','-frames:v','1',f'public/{name}.png'],check=True)
print(json.dumps(data,indent=2))
